x-pack/test_serverless/shared/services/user_manager/saml_auth.ts

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import axios, { AxiosResponse } from 'axios';
+import Url from 'url';
+import * as cheerio from 'cheerio';
+// import { createSAMLResponse as createFakeSAMLResponse } from 'kbn-mock-idp-plugin';
+
+export interface SAMLSessionParams {
+  username: string;
+  password: string;
+  kbnHost: string;
+  kbnVersion: string;
+}
+
+export interface FakeSAMLSessionParams {
+  username: string;
+  email: string;
+  fullName: string;
+  role: string;
+  kbnHost: string;
+}
+
+const envHosts: { [key: string]: string } = {
+  qa: 'console.qa.cld.elstc.co',
+  staging: 'staging.found.no',
+  production: 'api.elastic-cloud.com',
+};
+
+const getSidCookie = (cookies: string[]) => {
+  return cookies?.[0].toString().split(';')[0].split('sid=')[1] ?? '';
```

Comment:
__optional nit:__ you might want to use `'tough-cookie'` package to make parsing simpler (take a look at the examples in our integration test)

Reply:
Thanks! Indeed simpler